internal/controller/isbservicerollout_controller.go

```diff
@@ -601,3 +603,12 @@ func (r *ISBServiceRolloutReconciler) updateISBServiceRolloutStatusToFailed(ctx
 
 	return statusUpdateErr
 }
+
+func getChildrenResourcesHealthyCondition(conditions []metav1.Condition) *metav1.Condition {
```

Comment:
Why not simplify this to return true and false similar to pipeline?

Reply:
No technical reason, just had written this before implementing PipelineRollout approach. Will change to simplify.